pkg/middleware/jwt_auth.go

```diff
@@ -0,0 +1,160 @@
+package middleware
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ScaleFT/xjwt"
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/grafana/grafana/pkg/login"
+	m "github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+var remoteKeySet *xjwt.RemoteKeyset
+
+func JwtAuthInit() {
+	keyset, err := xjwt.NewRemoteKeyset(context.TODO(), xjwt.KeysetOptions{
```

Comment:
TODO should not be used. Why does NewRemoteKeyset need a context?

Reply:
@bergquist 
> Whats the reason for using ScaleFT/xjwt instead of more popular packages like https://github.com/dgrijalva/jwt-go ?

`https://github.com/dgrijalva/jwt-go` appears to be a library for parsing JWTs. The value that xjwt brings to the table is that it keeps a live cache of the JWKS that the proxy specifies as the keys used to validate the signed headers. xjwt is built on top of the https://github.com/square/go-jose library that implements JWT and JWKS support.

This is also why it needs a context, because it runs in a go routine periodically checking and making sure that it has the latest version of the JWKS published by the signer. Because the signer can rotate the keys used to sign headers on their own schedule, it is important that we always have an up to date copy. 

I chose `context.TODO()` just because I wasn't sure what context to use in this case, and was looking for guidance there. I'll double back and pass a context in to the `JwtAuthInit()` function I guess.